src/helpers.js

```diff
@@ -192,45 +192,57 @@ export function log(context, status, ...message){
 
 /**
  * Get JSON options
- * @returns void
+ * ID priority: data-attribute > selector#id > unique id
+ * @version 0.0.1
+ * @returns {object}
  */
-export function getOptions(context){
+export function getOptions(context, defaultOptions){
+    const numeric = ['duration', 'activeSection']; // convert these props to float
+    const wrapper = context.wrapper;
+
     // options from attribute
-    let string = context.wrapper.getAttribute(context._attr.container);
+    let dataAttribute = wrapper.getAttribute(context._attr.container);
     let options = {};
 
-    if(!isJSON(string)){
-        context.id = context.options.id;
-        return;
+    // data attribute doesn't exist or not JSON format -> string
+    if(!dataAttribute || !isJSON(dataAttribute)){
+        // reassign id
+        const id = dataAttribute || wrapper.id || defaultOptions.id;
```

Comment:
Fix me if I'm wrong, I don't see anywhere from the old version that we need `wrapper.id`, why do we need this here?

Reply:
I see that in `Easy Select` library, we respect `id` from the element's id so I want it to be consistent through our ES libraries. Or we don't need that, I will remove it if you want.